Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/nbclient/client.py", line 1314, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
from ngsolve import *
from netgen.occ import *
from time import time

from ngsolve.webgui import Draw
from netgen.webgui import Draw as DrawGeo

# box = Box((-1,-1,-1), (1,1,0))
# sp = Sphere((0.5,0,0), 0.2)
disc = Cylinder( Axes((0.5,0,0), X, Y), r=0.4, h=0.2)
shape = box-disc
geo = OCCGeometry(shape)

h = 0.1
        
mesh = Mesh( geo.GenerateMesh(maxh=h))
mesh.Curve(3)
Draw(mesh);
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
Cell [0;32mIn[1], line 11[0m
[1;32m      8[0m [38;5;66;03m# box = Box((-1,-1,-1), (1,1,0))[39;00m
[1;32m      9[0m [38;5;66;03m# sp = Sphere((0.5,0,0), 0.2)[39;00m
[1;32m     10[0m disc [38;5;241m=[39m Cylinder( Axes(([38;5;241m0.5[39m,[38;5;241m0[39m,[38;5;241m0[39m), X, Y), r[38;5;241m=[39m[38;5;241m0.4[39m, h[38;5;241m=[39m[38;5;241m0.2[39m)
[0;32m---> 11[0m shape [38;5;241m=[39m [43mbox[49m[38;5;241m-[39mdisc
[1;32m     12[0m geo [38;5;241m=[39m OCCGeometry(shape)
[1;32m     14[0m h [38;5;241m=[39m [38;5;241m0.1[39m

[0;31mNameError[0m: name 'box' is not defined

